ModelBundle/Repository/AbstractRepository.php

```diff
@@ -57,4 +61,145 @@ protected function hydrateAggregateQuery(Stage $qa, $elementName, $idSelector =
 
         return $contentCollection;
     }
+
+    /**
+     * @param Stage $qa
+     *
+     * @return int
+     */
+    protected function countDocumentAggregateQuery(Stage $qa){
+        $qa->group(array(
+            '_id' => null,
+            'count' => array('$sum' => 1)
+        ));
+        $res = $qa->getQuery()->aggregate();
+
+        return $res[0]['count'];
+    }
+
+    /**
+     * @param array|null   $columns
+     * @param string|null  $search
+     *
+     * @return Stage
+     */
+    protected function createAggregationQueryForFilterSearch($columns = null, $search = null){
+        $qa = $this->createAggregationQuery();
+
+        if( null !== $columns) {
+            $filterSearch = $this->generateFilterSearch($columns, $search);
+            if (null !== $filterSearch) {
+                $qa->match($filterSearch);
+            }
+        }
+
+        return $qa;
+    }
+
+    /**
+     * Create Query for paginate, order and filter by columns
+     *
+     * @param integer|null $limit
+     * @param integer|null $skip
+     * @param array|null   $columns
+     * @param array|null   $order
+     * @param string|null  $search
+     *
+     * @return Stage
+     */
+    protected function createAggregationQueryForPaginateAndSearch($limit = null, $skip = null, $columns = null, $order = null, $search = null)
```

Comment:
you should place the argument with the most used first.

I would more do  : `$colmuns, $search, $limit, $skip, $order`